plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -125,13 +127,8 @@ export const CatalogTable = ({
       const isStarred = isStarredEntity(rowData);
       return {
         cellStyle: { paddingLeft: '1em' },
-        icon: () =>
-          isStarred ? (
-            <Star htmlColor="#f3ba37" fontSize="small" />
-          ) : (
-            <StarOutline fontSize="small" />
-          ),
-        tooltip: isStarred ? 'Remove from favorites' : 'Add to favorites',
+        icon: () => favouriteEntityIcon(isStarred),
+        tooltip: favouriteEntityTooltip(isStarred),
```

Comment:
if I understand things correctly, you should be able to just do 

```
icon: () => <FavouriteEntity entity={rowData} />
```

and skip the tooltip since the icon itself has one already.

Reply:
Agree - this seems intentional. I tried it but then this happened: 

![bWFjNHiNIZ](https://user-images.githubusercontent.com/8904624/86368299-50c14a00-bc7d-11ea-9fc0-797b8e64f24e.gif)

Unfortunately the [material-table Actions do not accept components](https://material-table.com/#/docs/features/actions), but rather inject an icon (or react component) into a icon-button. Weird API, but this was my take to abstract the behavior in a new component & still use it as a material-table Action.

Reply:
Ah right! I should have known. Yeah go ahead, we can look into alternatives to that entire column or whatever at a later point